Examples/Scripts/SetUpHCPPipeline.sh

```diff
@@ -1,12 +1,27 @@
 #!/bin/echo This script should be sourced before calling a pipeline script, and should not be run directly:
 
+SAVEHCPPIPE="${HCPPIPEDIR:-}"
+
+# Set up specific environment variables for the HCP Pipeline
+export HCPPIPEDIR=
+
+#trick to allow certain setups to source an unmodified SetUp... and get some functionality
+if [[ -z "$HCPPIPEDIR" ]]
```

Comment:
Isn't this always going to be true, given that 3 lines earlier is `export HCPPIPEDIR=` ?

Reply:
That is the line the user is intended to edit.

Reply:
Ah, ok.  But I'd suggest that we consolidate all the variables that the user needs to set into one section at the top (HCPPIPEDIR, MSMBINDIR, MATLAB_COMPILER_RUNTIME) and then follow with all the stuff that the user shouldn't need to mess with, with a comment to that effect.  Right now it is a bit confusing how the `if [[ -z "$HCPPIPEDIR" ]]` stuff is "breaking up" the variables that the user needs to set.

Also, within that conditional, why aren't we using `export HCPPIPEDIR=` rather than just `HCPPIPEDIR=`?

Reply:
Since one of the default values (MSMCONFIGDIR) uses HCPPIPEDIR, they can't all be together if we have this conditional allowing an external value of HCPPIPEDIR to be used without editing.  Either these two variables are separated by some code, or we abandon this feature.

Exporting a variable once sets an internal flag on it.  Redundantly exporting has no effect.  The fact that it can be combined with setting a value is only for convenience.

Reply:
Another possibility is that if the user shouldn't actually have control of MSMCONFIGDIR, that can be moved to finish_hcpsetup.shlib.  However, this would mean that if someone wants to play with registration parameters, they have to make edits inside their repository.

Reply:
My point was that there is no need for the setting of MSMBINDIR, MATLAB_COMPILER_RUNTIME and FSL_FIXDIR to occur after the `if [[ -z "$HCPPIPEDIR" ]]` conditional, since none of those depend on `HCPPIPEDIR`.  So, I'd move up all the stuff that the user is expected to have to mess with, and then distinguish from the rest of the script with a :
```
# ----------------------
# Should not need to modify anthing beyond here
# ----------------------
```
section.

Now that you mention it, yes I think the setting of `MSMCONFIGDIR` should go into `finish_hcpsetup.shlib`, since very few people will ever need to mess with this, since we are including the config files as part of the repo itself.
If someone wants to mess with their own config files, they can "reset" `MSMCONFIGDIR` at the end of `SetUpHCPPipeline.sh`, or simply reset it outside of `SetUpHCPPipeline.sh` entirely.

My one minor concern about not "reexporting" `HCPPIPEDIR` is that if a user deletes the `export HCPPIPEDIR=` line entirely, then it potentially doesn't get exported at all, right?  Why not just redundantly reexport, since it causes no harm, and protects against users that might not know all the details of how exporting works?

Reply:
If they delete an export line entirely, they are doing something wrong.  I guess I can see the possibility.

There already is a line saying the user shouldn't need to modify anything below it.